src/gluonnlp/model/transformer.py

```diff
@@ -221,17 +234,26 @@ def __init__(self, attention_cell='multi_head', units=128,
                                  prefix='proj_')
             self.ffn = self._get_positionwise_ffn(use_bert_ffn, units, hidden_size, dropout,
                                                   use_residual, weight_initializer,
-                                                  bias_initializer)
-            self.layer_norm = _get_layer_norm(use_bert_layer_norm, units)
+                                                  bias_initializer, activation=activation,
+                                                  layer_norm_eps=layer_norm_eps)
+            self.layer_norm = _get_layer_norm(use_bert_layer_norm, units,
+                                              layer_norm_eps=layer_norm_eps)
 
     def _get_positionwise_ffn(self, use_bert, units, hidden_size, dropout, use_residual,
-                              weight_initializer, bias_initializer):
+                              weight_initializer, bias_initializer, activation=None,
+                              layer_norm_eps=None):
         from .bert import BERTPositionwiseFFN
         positionwise_ffn = BERTPositionwiseFFN if use_bert else PositionwiseFFN
-        return positionwise_ffn(units=units, hidden_size=hidden_size, dropout=dropout,
-                                use_residual=use_residual, weight_initializer=weight_initializer,
-                                bias_initializer=bias_initializer)
-
+        ffn_params = {'units':units, 'hidden_size':hidden_size, 'dropout':dropout,
+                      'use_residual':use_residual,
+                      'weight_initializer': weight_initializer,
+                      'bias_initializer': bias_initializer}
+        if use_bert:
```

Comment:
The fact that layer_norm_eps and activation only take effect if use_bert_ffn=True makes the code harder to maintain. Shall we add these two flags to PositionwiseFFN, too?

Reply:
Yes, added